lib/src/lib.rs

```diff
@@ -157,7 +157,7 @@ pub mod error {
 }
 
 /// An error originating from the SurrealDB client library
-#[derive(thiserror::Error, Debug)]
+#[derive(thiserror::Error, Debug, serde::Serialize)]
```

Comment:
Hi @hchockarprasad could you `use serde::Serialize` at the top of the file, and then just have:

```rs
#[derive(thiserror::Error, Debug, Serialize)]
```